### PR TITLE
Added in Passport Authentication, also added in all the changes that Nick & Kerrin posted about.

### PR DESCRIPTION
**Pull request by RipNasty**
- Updated usersRouter.js, the User model, and added in a UserProfile component.

### CODE REVIEW

**Comment by AngeC22** [comment]
we made it! looks good

**Comment by Chepenik** [comment]
Looks good!

File: client/src/components/UserProfile.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const UserProfile = ({ user }) => {
+    return (
+        <div>
+            <h1>This is a page built for a user to see</h1>
```
Comment (dovakeen118) [inline client/src/components/UserProfile.js:6]: we should avoid pages or implementations that are "placeholders". It's fine if more info will be added to this page overtime, but we need to think of the current feature this provides to users. Right now this does not show them any information. At the very least, this should start to display information about the user on their profile - like their email, etc.

**Review by dovakeen118 — COMMENTED** [review]
one small comment to make use of the Profile page component in a meaningful way

File: client/src/components/layout/TopBar.js [hunk 2]
```diff
@@ -28,6 +28,9 @@ const TopBar = ({ user }) => {
           <li>
             <Link to="/">Home</Link>
           </li>
+          <li>
+            <Link to="/profile">Your Profile</Link>
```
Comment (dovakeen118) [inline client/src/components/layout/TopBar.js:32]: if the profile is an authed route (meaning a user needs to be signed in) then this link should be included with the `authenticatedListItems` to ensure it only appears when a user is signed in

**Review by dovakeen118 — COMMENTED** [review]
tiny change for how the Profile route was added to the nav bar!

**Pull request merged**